### PR TITLE
Make terraform provider version a configurable property

### LINKED ISSUE #607: Make terraform provider version a configurable property
Currently Terraform provider versions of CSP is directly part of the Provider string. 
Eg - https://github.com/eclipse-xpanse/xpanse/blob/main/modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/providers/Openstack.java#L21

- [ ] Move version for each of the CSP to application.properties
- [ ] Document in README.md with a new section describing where the configuration exists and how to update it.

### PR DESCRIPTION
**Pull request by WangLiNaruto**
Fixed #607

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -238,6 +238,38 @@ If no AES private key file aes_sec is found in the intended location or if the f
 no encryption of the sensitive variables will take place. All data will stored in plain text within
 the JVM.
 
+### Version Configuration For Terraform Providers
+
+Xpanse currently uses Terraform to orchestrate resources on cloud providers, but versions of
+Terraform for cloud providers are in constant iteration. In order to configure the Terraform version
+of the cloud providers more conveniently and quickly, you can configure the Terraform version of the
+cloud providers in the xpanse configuration file `resource/application.properties`
```
Comment (swaroopar) [inline README.md:246]: please give exact link to this file in the repo. (you can link repo files easily in the readme doc)

File: README.md [hunk 2]
```diff
@@ -238,6 +238,38 @@ If no AES private key file aes_sec is found in the intended location or if the f
...
+
+#### The Version Key Of The Terraform Providers
+
+|  Terraform Providers   | key  |
+|  ----  | ----  |
+| AWS  | `terraform.privider.aws.version` |
```
Comment (swaroopar) [inline README.md:252]: privider -> typo. provider

File: README.md [hunk 3]
```diff
@@ -238,6 +238,38 @@ If no AES private key file aes_sec is found in the intended location or if the f
...
+| HuaweiCloud  | `terraform.privider.huawei.version` |
+| OpenStack  | `terraform.privider.openstack.version` |
+| FlexibleEngine  | `terraform.privider.flexibleengine.version` |
+
+#### How To Configure The Version Value Of The Terraform Providers
```
Comment (swaroopar) [inline README.md:257]: How To Configure The Version of Terraform Providers

File: README.md [hunk 4]
```diff
@@ -238,6 +238,38 @@ If no AES private key file aes_sec is found in the intended location or if the f
...
+`Version Constraints` in https://developer.hashicorp.com/terraform/language/providers/requirements#version-constraints
+
+
+The following are some examples, please refer to
+```properties
+terraform.provider.aws.version=~> 4.0
+terraform.provider.openstack.version=>= 1.48.0
+terraform.provider.flexibleengine.version=>= 1.30.0
+terraform.provider.huaweicloud.version=~> 1.51.0
+
```
Comment (swaroopar) [inline README.md:269]: pls remove these extra empty lines.

Comment (WangLiNaruto) [reply README.md:269]: ok

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**